File: lib/folavirt/networking/Response.py
#!/usr/bin/env python2
# -*- coding: utf-8 -*-

import json

class Response():
    def __init__(self, command = "", data = {}):
        self.command = command
        self.data = data
    
    errorcode = 0
    data = {}
    command = ""
    
    def setErrorCode(self, code):
        self.errorcode = code
    
    def getErrorCode(self):
        return self.errorcode
        
    def setData(self, array):
        self.data = array
        
    def getData(self):
        return self.data
        
    def setCommand(self, command):
        self.command = command
        
    def getCommand(self):
        return self.command
    
    def getJSON(self):
        retarray = {}
        retarray['errorcode'] = self.errorcode
        retarray['data'] = self.data
        retarray['command'] = self.command
        
        return json.dumps(retarray)
    
    def setFromJSON(self, jsondata):
        retarray = json.loads(jsondata)
        
        self.errorcode = retarray['errorcode']
        self.data = retarray['data']
        self.command = retarray['command']
    

File: www/module/Application/src/Application/Model/Folavirt/Networking/Response.php
<?php

namespace Application\Model\Folavirt\Networking;

/**
 * Odpowiedź agenta
 * 
 * @package Application\Model\Folavirt
 * @see lib/folavirt/networking/Response
 */
class Response
{
	public $command;
	public $data = array();
	public $errorcode = 0;
	
	/**
	 * Ustawia polecenie
	 * 
	 * @param string
	 * @return void
	 */
	public function setCommand($command)
	{
		$this -> command = $command;
	}
	
	/**
	 * Zwraca polecenie
	 * 
	 * @param void
	 * @return string
	 */
	public function getCommand()
	{
		return $this -> command;
	}
	
	/**
	 * Ustawia parametry
	 * 
	 * @param array
	 * @return void
	 */
	public function setData($data)
	{
		$this -> data = $data;
	}
	
	/**
	 * Zwraca parametry
	 * 
	 * @param void
	 * @return array
	 */
	public function getData()
	{
		return $this -> data;
	}
	
	/**
	 * Ustawia kod błędu
	 * 
	 * @param błąd
	 * @return void
	 */
	public function setErrorCode($error)
	{
		$this -> errorcode = $error;
	}
	
	/**
	 * Zwraca kod błędu
	 * 
	 * @param void
	 * @return błąd
	 */
	public function getErrorCode()
	{
		return $this -> errorcode;
	}
	
	/**
	 * Zwraca zapytania w formacie JSON
	 * 
	 * @param void
	 * @return json
	 */
	public function getJSON()
	{
		$retarray = array();
		$retarray['command'] = $this -> command;
		$retarray['data'] = $this -> data;
		$retarray['errorcode'] = $this -> errorcode;
		
		return json_encode($retarray);
	}
	
	/**
	 * Ustawia pola na podstawie ciągu JSON
	 * 
	 * @param json
	 * @return void
	 */
	public function setFromJSON($json)
	{
		$retarray = json_decode($json, 1);
		$this -> command = $retarray['command'];
		$this -> data = $retarray['data'];
		$this -> errorcode = $retarray['errorcode'];
	}
}
